File: app/src/main/java/com/example/vladimir/easyenglishlearn/word_selection/WordSelectionDto.java
package com.example.vladimir.easyenglishlearn.word_selection;

import com.example.vladimir.easyenglishlearn.Constants.Exercises;
import com.example.vladimir.easyenglishlearn.model.Word;

import java.util.ArrayList;

class WordSelectionDto {

    private boolean mTranslationDirection;
    private ArrayList<Word> mSelectedWordList;
    @Exercises
    private String mExercise;

    WordSelectionDto(boolean translationDirection,
                     ArrayList<Word> selectedWordList,
                     @Exercises String exercise) {
        mTranslationDirection = translationDirection;
        mSelectedWordList = selectedWordList;
        mExercise = exercise;
    }

    boolean isTranslationDirection() {
        return mTranslationDirection;
    }

    ArrayList<Word> getSelectedWordList() {
        return mSelectedWordList;
    }

    String getExercise() {
        return mExercise;
    }
}

File: app/src/main/java/com/vladimir_x/easyenglishlearn/ui/word_selection/WordSelectionDto.kt
package com.vladimir_x.easyenglishlearn.ui.word_selection

import com.vladimir_x.easyenglishlearn.Constants.Exercises
import com.vladimir_x.easyenglishlearn.model.Word
import com.vladimir_x.easyenglishlearn.ui.model.WordUI
import java.util.*

data class WordSelectionDto(
    val isTranslationDirection: Boolean,
    val selectedWordList: ArrayList<WordUI>,
    @Exercises val exercise: String
)